src/date-picker/embedded.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { DatePickerProps } from './interfaces';
+import Calendar from './calendar';
+import { memoizedDate } from './calendar/utils/date';
+import { useDatePicker } from './use-date-picker';
+
+export const DatePickerEmbedded = ({
+  value,
+  locale = '',
+  startOfWeek,
+  isDateEnabled,
+  nextMonthAriaLabel,
+  previousMonthAriaLabel,
+  todayAriaLabel,
+  calendarHasFocus,
+}: DatePickerEmbeddedProps) => {
+  const { displayedDate, selectedDate, onChangeMonthHandler, onSelectDateHandler } = useDatePicker({
```

Comment:
According to its signature, the useDatePicker takes value and onChange. I bet this embedded component is now readonly.

Reply:
Right, I'll need to add the `onChange` here (and make it required)